### ISSUE TITLE
Obsidian-style titles break link references

### ISSUE THREAD

**Issue by vJourneyman** [foambubble/foam#764]
- Foam version: 0.14.2
- Platform: Linux
- Issue occur on the [foam template](https://github.com/foambubble/foam-template) repo: Yes

**Summary**
When creating wikilinks to Obsidian-style-titled files with spaces in the filename, link reference definitions stop working. I believe this is because reference-style links require the link-target to conform to the standards of URLs (spaces rendered as + or %20). Or alternately surrounded by brackets. 

**Steps to reproduce**
1. Clone foam-template repo in GitHub.
2. Clone it locally
3. Change settings.json to include: 
```
"foam.edit.linkReferenceDefinitions": "on", 
"foam.edit.linkReferenceDefinitions": "withExtensions"
```
4. Create a new note, new-note1.md
5. Create a new note, new-note2.md
6. Create a new note, New Note 3.md
7. Create a new note, New Note 4.md
8. From new-note1.md, create a wikilink to new-note2.md; Confirm that the link extension definitions were created.
9. From new-note2.md create a wikilink to [[new-note1]] and to [[New Note 3]] ; Confirm that the link extension definitions were created. Confirm that the link to New Note 3.md is not enclosed in angle brackets
10. From new-note2.md create a standard-style link with angle brackets:   
```
[New Note 3](<New Note 3.md>)
```
No link reference should be created. Confirm that this link does not show up on the Foam Graph.
11. From new-note2.md create a reference-style link: 
```
[[New Note 4]]
[New Note 4]: <New Note 4.md> "New Note 4"
```
Make sure that your manual reference is before the automatically created block: [//begin]
Confirm that this link _does_ show up on the foam graph.
Unfortunately, a second automatic reference link will be created that won't work (we showed that with the links to New Note 3.md in steps 9 and 10).
12. From New Note 3.md create a wikilink to new-note1.md and new-note2.md ; Confirm that the link extension definitions were created.
13. Push to the cloned repo.
14. Inspect the rendering of the four new note files in GitHub. links to new-note1.md and new-note2.md will render correctly. Wiki-style link to New Note 3.md will not render but the standard-style link will. Confirm that the manually created reference link to New Note 4.md works.
15. Confirm that when viewing new-note2.md, the there is a broken and visible block at the bottom of the file which was supposed to be enclosed by [[//begin]] and [[//end]] and be hidden. 
16. Confirm that the rendering of new-note1.md without wikilinks to an Obsidian-style file renders correctly without the hidden block becoming visible.

**Additional information**
I think the fix is for auto-generated link references to enclose the link target in angle brackets if there's a space in the filename or directory name. Always enclosing it in angle brackets should work as well.

**Comment by vJourneyman** [comment]
Are the eyes emoji an indication that it's being looked at, or side-eye, like, "What is this guy even doing?" :-)

**Comment by riccardoferretti** [comment]
Looking ;) GH would have to be pretty mean for making the latter a default option ;)

**Comment by vJourneyman** [comment]
Essentially this boils down to this bracket notation with spaces in the titles working:
`[New Note 4]: <New Note 4.md> "New Note 4"`

And this notation with spaces in the titles not working:
`[New Note 4]: New Note 4.md "New Note 4"`

By not working, I mean not rendering properly in VS Code's markdown preview and at GitHub.

**Comment by LogeshG5** [comment]
Yes, if spaces are replaced by %20 in the link references static site generator (in my case, Docusaurus) is happy and linking properly.
